src/classes/BDI_ManageAdvancedMappingController.cls

```diff
@@ -0,0 +1,51 @@
+/*
+    Copyright (c) 2017, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2020
+* @description Controller used by the BDI_ManageAdvancedMapping component
+*/
+public with sharing class BDI_ManageAdvancedMappingController {
+
+    /**
+     * @description Checks if the running user has system admin permissions
+     * @return Boolean
+    */
+    @AuraEnabled
+    public static Boolean isAdminUser(Id userId) {
+        List<User> runningUser = [SELECT ProfileId FROM User WHERE Id = :userId];
```

Comment:
Should we start adding perm checks for methods that run a query? Something like `UTIL_Permissions.getInstance().canRead(User.SObjectType)` perhaps

Reply:
I think in general, yes we should include WITH SECURITY_ENFORCED - however this is an interesting one since we are actually checking permissions with this query so throwing a security error here seems almost redundant

Reply:
We could also refactor this with UserInfo.getProfileId() to avoid doing a SOQL query at all but that might be overkill

Reply:
@ptewson-sfdo The reason for the query in this case is to get the profileId but also to get the `User` SObject record since that's what has to be passed as a parameter to the `isSysAdmin` method in the user service.

Reply:
@ptewson-sfdo I've added the security check to this query. I figure the user should always have access to their own user record, but this should at least prevent the query from being flagged in a Checkmarx, PMD, or other security scan.